### ISSUE TITLE
Docker composer up bug

### ISSUE THREAD

**Issue by yarekc** [pH7Software/pH7-Social-Dating-CMS#554]
Same issue that here: https://github.com/pH7Software/pH7-Social-Dating-CMS/issues/515

On a brand new Debian distribution

> docker-compose up

```
Starting ph7_db_1
Creating ph7_php_1

ERROR: for php  Cannot create container for service php: invalid mode: /var/www/
ERROR: Encountered errors while bringing up the project.
```

**Comment by pH-7** [comment]
Hey @yarekc 
Thanks a lot for this!
Do you mind to try with those changes done here https://github.com/pH7Software/pH7-Social-Dating-CMS/pull/472 ?

**Comment by yarekc** [comment]
Sorry, 
Iread #472 and this is all very unclear. 
- Is there a working docker-compose.yml somwhere ?
- any chance to get a working image on dockerhub ?

I recommend you test the docker-compose file on a totally new/blank server
(I am not a docker expert, but I did not have any problems with WP or JOOMLA)

**Comment by polynamaude** [comment]
@yarekc At first, what version of docker are you using ? The ones packaged with Debian (as with most distribution) is quite old when compared to the versions rolling out. At first, install the latest stable version as a precompiled binary.

Also, don't make comparison like "I can run a container of WP/Joomla but not pH7CMS" there's no use for such and the only thing it shows is that you can indeed run the **_docker_** command.

If you read the thread relating to issue to #472 then you'll see that the person implementing the docker files didn't seem to have finished her work.

We all know things like "Try on a clean server", the question is more when ? We are all volunteer here.

**Comment by yarekc** [comment]
I run the latest one: 
Docker version 18.09.6, build 481bc77

this comparaison was just to tell you: I am not a docker expert, but I  have enough knowledge to run other docker images correctly.

So I still believe, there is a problem with the docker-composer file

Steps to reproduce the error:
- clean latest Debian9
- latest docker 
- copy the docker-composer.yml  from the git
- run it with :  `docker-compose up`

**Comment by polynamaude** [comment]
@yarekc Yes there's possibly still problem with the docker files. It's the exact reason why #472 was opened.

**Comment by pH-7** [comment]
@yarekc Did you check/ask the error at Stackoverflow? You might get the right fix quickly 💪

**Comment by mattthhdp** [comment]
i too want to know how to, for testing purpose.
the #472 dosen't fix the issus for me, i also try to build the docker image but there is error there too ...
Any idea? i am willing to test.

**Comment by NicolasCARPi** [comment]
This patch fixes the issue:

~~~patch
diff --git a/docker-compose.yml b/docker-compose.yml
index 84ba358c4..261a949f0 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -6,7 +6,7 @@ services:
         ports:
             - "8080:80"
         volumes:
-            - .:/var/www:/var/www/
+            - .:/var/www/
             - ./sites:/etc/nginx/conf.d
             - ./nginx.conf:/etc/nginx/ph7cms.conf
         depends_on:
@@ -16,7 +16,7 @@ services:
     php:
         image: php:7.2-fpm
         volumes:
-            - .:/var/www:/var/www/
+            - .:/var/www/
         command: bash -c "composer install"
         working_dir: /var/www:/var/www/
     db:
~~~

But to be honest, the whole Docker aspect of this project needs some love. I was editing the Dockerfile (to improve the way composer is installed), but quickly realized there was a lot of things wrong (the list is long but the main point is: it would not work). And I saw the docker-compose.yml file not using this Dockerfile anyway.

@pH-7 you might want to remove the Dockerfile from the repo, as there is no local build of it in the docker-compose.yml file (also it's full of errors).

Note that the patch I provided will allow launching the containers, but they fail because there are still a lot of things wrong in the current implementation.

My advice is to start from scratch regarding the whole docker aspect ;).

**Comment by pH-7** [comment]
Thanks for reporting this.

This is now fixed in 18.x. Docker Compose mount/working-dir syntax has been corrected, including the invalid volume mode pattern and DB port mapping, so containers can start correctly.

Take care and thank you for using pH7 Social Dating Builder,

[Pierre-Henry](https://github.com/pH-7)
